viewer/packages/3d-overlays/src/OverlayPointsObject.ts

```diff
@@ -23,87 +22,114 @@ import {
     this._positionAttribute = new BufferAttribute(this._positionBuffer, 3);
+    this._colorBuffer = new Float32Array(maxNumberOfPoints * 3).fill(1);
+    this._colorAttribute = new BufferAttribute(this._colorBuffer, 3);
     geometry.setAttribute('position', this._positionAttribute);
+    geometry.setAttribute('color', this._colorAttribute);
     geometry.setDrawRange(0, 0);
 
-    const frontMaterial = this.createIconsMaterial(
+    const {
       spriteTexture,
-      1,
-      LessEqualDepth,
       minPixelSize,
       maxPixelSize,
       radius,
-      colorTint
-    );
-    const backMaterial = this.createIconsMaterial(
+      colorTint = new Color(1, 1, 1),
+      depthMode = LessEqualDepth,
+      collectionOpacity = 1
+    } = materialParameters;
+
+    const frontMaterial = this.createIconsMaterial(
       spriteTexture,
-      0.5,
-      GreaterDepth,
+      collectionOpacity,
+      depthMode,
       minPixelSize,
       maxPixelSize,
       radius,
       colorTint
     );
-    const [frontPoints, backPoints] = this.initializePoints(geometry, frontMaterial, backMaterial);
+    const frontPoints = this.initializePoints(geometry, frontMaterial);
     this.add(frontPoints);
-    this.add(backPoints);
 
     this._geometry = geometry;
     this._frontMaterial = frontMaterial;
-    this._backMaterial = backMaterial;
   }
 
-  public setPoints(points: Vector3[]): void {
-    points.forEach((point, index) => {
-      this._positionBuffer[index * 3 + 0] = point.x;
-      this._positionBuffer[index * 3 + 1] = point.y;
-      this._positionBuffer[index * 3 + 2] = point.z;
-    });
+  public setPoints(points: Vector3[], colors?: Color[]): void {
+    if (colors && points.length !== colors?.length)
+      throw new Error('Points positions and colors arrays must have the same length');
+
+    for (let index = 0; index < points.length; index++) {
+      this._positionBuffer[index * 3 + 0] = points[index].x;
+      this._positionBuffer[index * 3 + 1] = points[index].y;
+      this._positionBuffer[index * 3 + 2] = points[index].z;
+
+      if (colors) {
+        this._colorBuffer[index * 3 + 0] = colors[index].r;
+        this._colorBuffer[index * 3 + 1] = colors[index].g;
+        this._colorBuffer[index * 3 + 2] = colors[index].b;
+      }
+    }
+
     this._positionAttribute.updateRange = { offset: 0, count: points.length * 3 };
     this._positionAttribute.needsUpdate = true;
+    this._colorAttribute.updateRange = { offset: 0, count: points.length * 3 };
+    this._colorAttribute.needsUpdate = true;
     this._geometry.setDrawRange(0, points.length);
+
+    this._geometry.computeBoundingBox();
+    this._geometry.computeBoundingSphere();
+  }
+
+  public addPoints(points: Vector3[]): void {
+    const lastDrawIndex = this._geometry.drawRange.count * 3;
+
+    if (lastDrawIndex + points.length * 3 > this._positionBuffer.length) return;
+
+    points.forEach((point, index) => {
```

Comment:
This method is not used anywhere and was added initially to be used when adding new points instead of setting the whole buffer again but it seems to work fine as is, so will remove it.